packages/docker-node/src/ckb-bin-network.ts

```diff
@@ -0,0 +1,259 @@
+import { Address, CellDep, Indexer, RPC, Transaction, commons, config, hd, helpers } from '@ckb-lumos/lumos'
+import { CKBNode } from './interface'
+import type { BinNodeStartOptions, BinNodeStopOptions, DeployOptions, InfraScript } from './types'
+import { spawn, execSync } from 'child_process'
+import { generateDevConfig } from './helper'
+import fs from 'fs'
+import path from 'node:path'
+import { waitUntilCommitted } from '@ckb-js/kuai-common'
+
+export class CKBBinNetwork implements CKBNode {
+  #port = '8114'
+  #host = '127.0.0.1'
+  #lumosConfig: config.Config = config.getConfig()
+
+  stop({ ckbPath, clear }: BinNodeStopOptions): void {
+    if (fs.existsSync(path.resolve(ckbPath, 'pid'))) {
+      const indexer = fs.readFileSync(path.resolve(ckbPath, 'pid', 'indexer'), 'utf-8')
+      spawn('kill', ['-9', indexer])
+      const miner = fs.readFileSync(path.resolve(ckbPath, 'pid', 'miner'), 'utf-8')
+      spawn('kill', ['-9', miner])
+    }
+
+    if (clear) {
+      fs.rmSync(path.resolve(ckbPath, 'ckb-miner.toml'))
+      fs.rmSync(path.resolve(ckbPath, 'ckb.toml'))
+      fs.rmSync(path.resolve(ckbPath, 'data'), { recursive: true, force: true })
+      fs.rmSync(path.resolve(ckbPath, 'default.db-options'))
+      fs.rmSync(path.resolve(ckbPath, 'specs'), { recursive: true, force: true })
+      fs.rmSync(path.resolve(ckbPath, 'pid'), { recursive: true, force: true })
+      fs.rmSync(path.resolve(ckbPath, 'dev.toml'))
+    }
+  }
+
+  #initConfig(ckbPath: string, genesisAccountArgs?: string[]) {
+    const devConfig = generateDevConfig(genesisAccountArgs)
+    fs.writeFileSync(path.resolve(ckbPath, 'dev.toml'), devConfig, { flag: 'w' })
+
+    if (!fs.existsSync(path.resolve(ckbPath, 'ckb.toml'))) {
+      const out = execSync(
+        `${path.resolve(ckbPath, 'ckb')} init -C ${ckbPath} --chain dev --import-spec ${path.resolve(
+          ckbPath,
+          'dev.toml',
+        )} --ba-arg 0x839f6f4d6fdf773d3e015f8b19fe5c4ccb07723d --ba-code-hash 0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8 --ba-hash-type type --ba-message 0x1234`,
+      )
+
+      console.info(out.toString())
+
+      fs.copyFileSync(path.resolve(__dirname, '../ckb/ckb-miner.toml'), path.join(ckbPath, 'ckb-miner.toml'))
+    }
+
+    if (!fs.existsSync(path.join(ckbPath, 'pid'))) {
+      fs.mkdirSync(path.join(ckbPath, 'pid'))
+    }
+  }
+
+  async #startIndexer(ckbPath: string, detached: boolean) {
```

Comment:
IMO, we can add an indexer parameter to start the node with indexer or without indexer
```suggestion
  async #startNode(ckbPath: string, detached: boolean, indexer: boolean = true) {
```